src/internal/common/maps/keyset.go

```diff
@@ -0,0 +1,21 @@
+package maps
```

Comment:
as a warning, this may collide with the golang `maps` package, which we use elsewhere.  I don't know a better name off the top of my head.  Maybe `keys`?  Then you can change the value from `keys.KeySet` to just Set, for `keys.Set`?